content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+  BufferedReader bufferedReader = null;  
+  BufferedWriter bufferedWriter = null;  
+  
+try {  
+socket = new Socket("localhost", 5000);  
+inputStreamReader = new InputStreamReader(socket.getInputStream());  
+outputStreamWriter = new OutputStreamWriter(socket.getOutputStream());  
+
+bufferedReader = new BufferedReader(inputStreamReader);  
+bufferedWriter = new BufferedWriter(outputStreamWriter);  
+
+Scanner scanner = new Scanner(System.in);  
+
+while (true)
+ {  
+  String msgToSend = scanner.nextLine();  
+
+  bufferedWriter.write(msgToSend);  
+
+  bufferedWriter.newLine();  
+
+  bufferedWriter.flush();  
+  
+  System.out.println("Server: " + bufferedReader.readLine());  
+  
+if (msgToSend.equalsIgnoreCase("BYE"))  
+    break;  
+  }  
+} catch (IOException e) {  
+   e.printStackTrace();  
+} finally {  
+try {  
+
+if (socket != null)  
+ socket.close();  
+if (inputStreamReader != null)  
+  inputStreamReader.close();  
+if (outputStreamWriter != null)  
+ outputStreamWriter.close();  
+if (bufferedReader != null)  
+ bufferedReader.close();  
+if (bufferedWriter != null)  
+ bufferedWriter.close();  
+} catch (IOException e) {  
+  e.printStackTrace();  
+          }  
+        }  
+    }  
+}
+```
+The output of the code:
+
+```
+Hey there 
+Server:  Message Received
+am doing this for the first time
+Server:  Message Received
+BYE
+Server:  Message Received
+
+```
+### Procedure of How to run the two programs if using Intellij or any other IDE
+1. Compile the two 
+2. First execute the program
+3. on the server **run**, then run the program client **run* 
+4. Type client window messages that are simultaneously received and shown by the Server window.
+5. Type **BYE** to end.
+
+### If using Command prompt/Terminal
+1. Depending on how you stored it, start the server program as the java server.
+2. Then there are two. Run the client application on a terminal like a java customer.
+3. Then Type messages in the client window
+
+Example 
+```
+Hey there 
+Server:  Message Received
+am doing this for the first time
+Server:  Message Received
+BYE
+Server:  Message Received
+```
+4.The server receives and simultaneously shows;
+
+```bash
+Connected
+Client: Hey there 
+Client: am doing this for the first time
+Client: BYE
+Closing connection
+```
+ Important note
+Sending "BYE" closes the connection between the Client and the Server.
+Port number may fail to work you can always change it and use a different one from those commercially owned.
+
+### Conclusion 
+
+In this article, we have been able to get an insight of sockets and TCP/IP protocol. We have understood what socket programming entails and its relation to Java. We have also seen a clear picture of client/server relationship, how server accepts information from client and sends it back.
+I urge the reader to use the knowledge learned above to write programs on how server accepts the clients' request in java programming language.
```

Comment:
> I urge the reader to use the knowledge learned above to write programs on how server accepts the clients' request in java programming language.

I would shorten this to something like: "I urge the reader to use the knowledge learned above to write networking programs in the java programming language"